### DISCUSSION TITLE
How do I implement a slider into my project?

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5816] [Help]
I need to insert a slider into my project.  I have never worked with sliders before.  I do not even know what field type it should be, let alone where to start to program one in Serenity.  I have found them under 'Themes' in the online demo, but cannot find the files on git.

I need to add a slider that represents values 1 to 5 for a survey response.
eg:
How confident do you feel using the system in the following transactions?

Search:   (slider response with values of 1 to 5)

So far I have this.
![image](https://user-images.githubusercontent.com/47765946/123049092-84732200-d442-11eb-9864-a8d9d669e231.png)

Row.cs
`        [DisplayName("Outcome 1"), NotNull, QuickSearch]
        public String Outcome1
        {
            get => fields.Outcome1[this];
            set => fields.Outcome1[this] = value;
        }`

Form.cs
`        [DisplayName("Search")]
        [LabelWidth("300")]
        public String Outcome1 { get; set; }`

When I implement the following in a SurveyClericalDialog.ts.html I get an error.
```
`<div class="d-flex justify-content-center my-4">
    <label>
        How confident do you feel using the system in the following transactions:
    </label>
    <div class="col-sm-6">
        <input id="Outcome1" type="text" name="Outcome1" value="">
    </div>
</div>
<script>
    $("#Outcome1").ionRangeSlider({
        min: 1,
        max: 5,
        type: 'single',
        step: 1,
//        postfix: "",
        prettify: false,
        hasGrid: true
    });
</script>`
```

I am probably doing it all wrong, so all help is greatly appreciated.

**Comment by VictorTomaili** [comment]
Check this howto guide: https://serenity.is/docs/howto/how_to_use_a_third_party_plugin_with_serenity

**Reply by shortsam1** [reply]
Thanks Victor.  I am working through this guide. 

However I cannot seem to work out these get and set etc; statements. I'm not sure what to change the '.value' to on the end of these.  I have looked at everything.  Keep in mind that I am looking at someone else's solution and trying to modify/implement it to my needs and I am a newbie to this.
![image](https://user-images.githubusercontent.com/47765946/123185839-2a25a000-d4da-11eb-9180-276e8053db26.png)

**Reply by shortsam1** [reply]
@Estrusco are you able to help?

I have changed the get/set stuff to this but still have one issue.
![image](https://user-images.githubusercontent.com/47765946/123186641-ed5aa880-d4db-11eb-985a-18688b080352.png)
````
namespace SparcOnlineForms {

    /**
     * This is an Slider editor widget
     *  
     */
    @Serenity.Decorators.element("<div/>")
    @Serenity.Decorators.registerEditor([Serenity.IGetEditValue, Serenity.ISetEditValue])
    export class SliderEditor extends Serenity.Widget<SliderEditorOptions>
        implements Serenity.IGetEditValue, Serenity.ISetEditValue {

        constructor(container: JQuery, options: SliderEditorOptions) {
            super(container, options);

            this.updateElementContent();
        }

        private updateElementContent() {
            var ticks: string;
            var tickslabels: string;

            // Here you can dynamically create ticks and ticks labels
            //for (var t of this.options.ticks) {
            //    [...]
            //}

            var elID = 'sld' + this.uniqueName;

            this.element.append('<input id="' + elID + '" />');
            var input = this.element.find('#' + elID);
            input.attr({
                'data-provide': 'slider',
                'data-slider-ticks': '[1, 2, 3, 4, 5]', // TODO: create dynamically
                //'data-slider-ticks-labels': '["One", "Two", "Three", "Four", "Five"]', // TODO: create dynamically
                'data-slider-min': 1, //this.options.min,
                'data-slider-max': 5, //this.options.max,
                'data-slider-step': 1, //this.options.step,
                'data-slider-value': 1//,
                //'data-slider-tooltip': 'hide'
            });
            input.append('<script>$("#' + elID + '").slider();</script>');
        }

        // Implement get and set
        public get value(): number {
            var sld = this.element.find('#sld' + this.uniqueName);
            return Q.toId(sld.slider('getValue'));
        }

        public set value(value: number) {
            var sld = this.element.find('#sld' + this.uniqueName);
            sld.slider('setValue', value);
        }

        public getEditValue(property, target) {
            target[property.name] = this.value;
        }

        public setEditValue(source, property) {
            this.value = source[property.name];
        }
    }

    export interface SliderEditorOptions {
        min: number;
        max: number;
        step: number;
        ticks: number[];
        tickslabels: string[];
    }
}
````

**Reply by VictorTomaili** [reply]
`(this.element[0].children[1] as any).value` must work. But ensure about children element must a value property.
Check your slider type definition about which type of parameter It's accepting. `sld.slider('setValue', value);` maybe this value must be a string value. you can use `toString()` or `as any`
